### DISCUSSION TITLE
Weathermap plugin scroll doesn't work on first click

### DISCUSSION THREAD

**Discussion by mahavirjain98** [thurban/plugin_weathermap#93] [Q&A] [answered]
When we click on weathermap tab in cacti, scrollbar is not visible at first if there are multiple maps, weathermap needs to be clicked again to see the scrollbar to scroll down to required maps.

**Comment by TheWitness** [comment]
Can you upload and animated gif of the issue?  Also, can you please look for JavaScript errors?

**Reply by mahavirjain98** [reply]
Don't see any javascript related errors/warnings.
Have tested with these browsers: firefox, edge, chrome, issues stays same.
Have clipped graph thumbnnails in middle for privacy protection:

**Below snip shows No scrollbar seen when weathemap tab is clicked first time:**
![snip_when_weathermap_dialog_opened_first_time](https://user-images.githubusercontent.com/36436460/220836628-0291c544-9916-4108-b729-bb38e8d96e7c.png)

**Below snip show scrollbar when either browser is refreshed or weathermap tab is clicked once-again:**
![snip_when_either_browser_is_refreshed_or_when_weathermap_tab_is_clicked_again](https://user-images.githubusercontent.com/36436460/220836659-9796cfb9-54b0-4c06-8c38-4ccabd815d59.png)

**Reply by TheWitness** [reply]
Hmm, the fact that it fixes itself after refresh has me scratching my head.  If you can send the animated gifs to my email, showing the debug console open that would help.

Do you have any other plugins loaded?  If so, name them.

TheWitness at cacti dot net.

**Reply by mahavirjain98** [reply]
Please find below snippet of error I saw in debugs.

![Screenshot 2023-02-24 153543](https://user-images.githubusercontent.com/36436460/221154717-6c53d644-fcbd-48fb-9c2d-a4808b3a1133.png)

I managed to fix the issue though, complete path to weathermap.css was missing in file named: weathermap-cacti-plugin.php.

I replaced all the occurrences of || **href="css/weathermap.css"/>';** || with || **href="' . $config['url_path'] . 'plugins/weathermap/css/weathermap.css"/>';** ||

Pasting the diff of the changes done, please confirm if the fix done to solve my problem was correct.

```
weathermap]# diff weathermap-cacti-plugin.php weathermap-cacti-plugin.php.old   
267c267
<                       print '<link rel="stylesheet" type="text/css" media="screen" href="' . $config['url_path'] . 'plugins/weathermap/css/weathermap.css"/>';
---
>                       print '<link rel="stylesheet" type="text/css" media="screen" href="css/weathermap.css"/>';
275c275
<               print '<link rel="stylesheet" type="text/css" media="screen" href="' . $config['url_path'] . 'plugins/weathermap/css/weathermap.css"/>';
---
>               print '<link rel="stylesheet" type="text/css" media="screen" href="css/weathermap.css"/>';
298c298
<               print '<link rel="stylesheet" type="text/css" media="screen" href="' . $config['url_path'] . 'plugins/weathermap/css/weathermap.css"/>';
---
>               print '<link rel="stylesheet" type="text/css" media="screen" href="css/weathermap.css"/>';
319c319
<               print '<link rel="stylesheet" type="text/css" media="screen" href="' . $config['url_path'] . 'plugins/weathermap/css/weathermap.css"/>';
---
>               print '<link rel="stylesheet" type="text/css" media="screen" href="css/weathermap.css"/>';

```

**Reply by TheWitness** [reply]
Good catch!  Can you do a pull request?

**Reply by mahavirjain98** [reply]
I am new to git and not aware about the process to do it.

**Reply by mahavirjain98** [reply]
Initiated pull request check if done correctly

**Reply by TheWitness** [reply]
Great!  Welcome to the team then!

**Comment by TheWitness** [comment, marked as answer]
Fixed by the users kind pull request.  Thanks!
